hibernate-core/src/main/java/org/hibernate/dialect/MySQLDialect.java

```diff
@@ -212,12 +214,12 @@ protected static DatabaseVersion createVersion(DialectResolutionInfo info) {
 	protected static DatabaseVersion createVersion(DialectResolutionInfo info, DatabaseVersion defaultVersion) {
 		final String versionString = info.getDatabaseVersion();
 		if ( versionString != null ) {
-			final String[] components = versionString.split( "\\." );
-			if ( components.length >= 3 ) {
+			final Matcher matcher = Pattern.compile("^(\\d+)\\.(\\d+)\\.(\\d+).*").matcher(versionString);
+			if ( matcher.matches() && matcher.groupCount() >= 3  ) {
```

Comment:
I'm thinking it would be simpler to just use one of the `split()` methods on `org.hibernate.internal.util.StringHelper`. Something like `StringHelper.split(".-",versionString)`.

WDYT?

Reply:
Yeah I get that but unless we actually have known cases where MySQL gives us something other than `x.y.x-blah`, I would not try to overgeneralize.